coverage_config_x86_64.json

```diff
@@ -1,5 +1,5 @@
 {
-  "coverage_score": 88.6,
+  "coverage_score": 87.2,
```

Comment:
The coverage is dropping here quite a bit. Do we know what's getting missed in the new paths?

Reply:
Good catch! I ran the coverage test locally and looked through the report. There are 4 functions in QueueSync added that are not covered by unit tests:
- desc_table
- avail_ring
- used_ring
- event_idx_enabled
These are getters that are calling into the `Queue` interface which is already covered by unit tests. For redundancy, I'll add some tests for these ones as well.

Then, there are also some new error types defined in this PR for which we don't have unit tests, so that's also contributing to the drop.

Besides these functions, I also deleted a lot of code that was previously covered by unit tests, so presumably that's also part of the reason why we're seeing this drop.

Reply:
Otherwise it looks pretty good to me. I'm looking at the vhost-user-backend work in progress now. The remaining build errors seems to be all about satisfying trait bounds so I'll need to get my "thinking like the compiler" hat on ;-)

Reply:
I think the problem is how I added the `mem` field in fact. I might be able to think about it a bit more next week if nobody else has time to pick it up.

Reply:
@stsquad I added unit tests for the new getters and setters and now the coverage is up to 88.5.

Reply:
@andreeaflorescu @stsquad please let me know when you'll have the `vhost-user-backend` PR ready so that I can fully port Cloud Hypervisor to this new version of `virtio-queue` and run the proper testing on it.

Reply:
@sboeuf I managed to integrate the changes in vhost-user-backend: https://github.com/andreeaflorescu/vhost-user-backend/tree/update_virtio_queue. I've also opened a PR: https://github.com/rust-vmm/vhost-user-backend/pull/78

Reply:
Cool! I'll have a try then!